lib/sidekiq/middleware/server/retry_jobs.rb

```diff
@@ -133,7 +133,9 @@ def attempt_retry(worker, msg, queue, exception)
             retries_exhausted(worker, msg)
           end
 
-          raise exception
+          if !msg['raise_on_dead'] || count >= max_retry_attempts
```

Comment:
There's probably a better flag name. I couldn't think of one short of `raise_on_death`, or maybe `exception_on_death`?